lib/tests/streamlit/scriptrunner/script_runner_test.py

```diff
@@ -184,7 +184,14 @@ def test_stop_script(self):
 
         time.sleep(0.1)
         scriptrunner.enqueue_rerun()
-        time.sleep(0.1)
+
+        # This test will fail if the script runner does not execute the infinite
+        # script's write call at least once during the final script run.
+        # The script runs forever, and when we enqueue a rerun it forcibly
+        # stops execution and runs some cleanup. If we do not wait for the
+        # forced GC to finish, the script won't start running before we stop
+        # the script runner, so the expected delta is never created.
+        time.sleep(0.3)
```

Comment:
Is `0.3` chosen arbitrary? Can race condition still happen?

Reply:
Yes, 0.3 is arbitrary, and the race condition could still happen. I thought about trying to rewrite it to not potentially have this problem, but since we don't have an api for blocking until the script runner is in a certain state, it seemed like a lot of effort, and I'm hoping that with the comment any future failures like this would be trivial to fix.